crates/rsonpath-lib/src/query/nonnegative_array_index.rs

```diff
@@ -0,0 +1,106 @@
+use super::error::ArrayIndexError;
+use std::fmt::{self, Display, Formatter};
+
+/// Array index to search for in a JSON document.
+///
+/// Represents a specific location from the front of the list in a json array.
+/// Provides the [IETF-conforming index value](https://www.rfc-editor.org/rfc/rfc7493.html#section-2).  Values are \[0, (2^53)-1].
+/// # Examples
+///
+/// ```
+/// # use rsonpath_lib::query::NonNegativeArrayIndex;
+///
+/// let idx = NonNegativeArrayIndex::new(2);
+///
+/// assert_eq!(idx.get_index(), 2);
+/// ```
+#[derive(Clone, Copy, PartialEq, Eq, Debug, PartialOrd, Ord)]
+pub struct NonNegativeArrayIndex(u64);
+
+/// The upper inclusive bound on index values.
+pub(crate) const ARRAY_INDEX_ULIMIT: u64 = (1 << 53) - 1;
```

Comment:
Does this need to be `crate` public?

Reply:
No.  However, I've refactored this to be a constant in the same way as ZERO now is.  So those are now both `pub`.